service/src/main/java/uk/nhs/adaptors/gpc/consumer/sds/builder/SdsRequestBuilder.java

```diff
@@ -97,21 +104,45 @@ public RequestHeadersSpec<?> buildRetrieveDocumentAsDeviceRequest(String fromOds
         return buildAsDeviceRequest(fromOdsCode, RETRIEVE_DOCUMENT_INTERACTION, correlationId);
     }
 
-    private RequestHeadersSpec<? extends RequestHeadersSpec<?>> buildMhsEndpointRequest(String odsCode, String interaction,
-        String correlationId) {
-        return buildClientFor(odsCode, interaction, correlationId, ENDPOINT_MHS_ENDPOINT);
+    private RequestHeadersSpec<? extends RequestHeadersSpec<?>> buildMhsEndpointRequest(String consumerOrgOdsCode,
+                                                                                        String interaction, String correlationId) {
+        return buildClientFor(consumerOrgOdsCode, interaction, correlationId, ENDPOINT_MHS_ENDPOINT);
     }
 
-    private RequestHeadersSpec<? extends RequestHeadersSpec<?>> buildAsDeviceRequest(String odsCode, String interaction,
-        String correlationId) {
+    private RequestHeadersSpec<? extends RequestHeadersSpec<?>> buildAsDeviceRequest(String odsCode,
+                                                                                     String interaction, String correlationId) {
         return buildClientFor(odsCode, interaction, correlationId, ENDPOINT_AS_DEVICE);
     }
 
+    public RequestHeadersSpec<? extends RequestHeadersSpec<?>> buildAsDeviceAsidRequest(String odsCode, String supplierOdsCode,
+                                                                                        String interaction, String correlationId) {
+        return buildAsidClientFor(odsCode, supplierOdsCode, interaction, correlationId, ENDPOINT_AS_DEVICE);
+    }
+
+    @NotNull
+    private RequestHeadersSpec<? extends RequestHeadersSpec<?>> buildAsidClientFor(String odsCode, String supplierOdsCode,
```

Comment:
We use the term `odsCode` in this method, and `consumerOrgOdsCode` elsewhere, perhaps we should consolidate on one name for this concept?